toolchain/install/BUILD

```diff
@@ -157,3 +160,62 @@ cc_test(
         "@llvm-project//llvm:Support",
     ],
 )
+
+# Build rules to construct packaged versions of the toolchain's install.
+
+pkg_files(
+    name = "pkg_exe_files",
+    # We break out the driver and LLD here because we can't easily use file
+    # groups that contain symlinks, so those are manually handled below. Other
+    # file groups should be directly included.
+    srcs = [
+        "prefix_root/bin/carbon",
+        "prefix_root/lib/carbon/llvm/bin/lld",
+    ],
+    attributes = pkg_attributes(mode = "0755"),
+    strip_prefix = strip_prefix.from_pkg("prefix_root"),
+)
+
+# Currently, `rules_pkg` can't replicate symlinks from the main tree. To an
+# extent, this is reasonable because we want to be much more explicit about the
+# symlink structure in the package where as for the filegroups we're comfortable
+# with whatever "just works" for development and testing.
+[
+    pkg_mklink(
+        name = "pkg_link_lld_alias_" + bin_name,
+        link_name = "lib/carbon/llvm/bin/" + bin_name,
+        target = "lld",
+    )
+    for bin_name in lld_bin_names
+]
+
+pkg_filegroup(
+    name = "pkg_files",
+    srcs = [
+        ":pkg_exe_files",
+    ] + [
+        ":pkg_link_lld_alias_" + bin_name
+        for bin_name in lld_bin_names
+    ],
+)
+
+# TODO: We should add support for injecting a version string into both the
+# output filename and the package directory name.
+pkg_tar(
+    name = "pkg_tar",
```

Comment:
Had you considered keeping target names distinct from rule names? e.g., something like "carbon_toolchain.txz.rule" for this and `pkg_zip`, or just removing the `pkg_` prefixes on all of these? (if just removing `pkg_` bugs you for collision risk, you could make a `pkg` subdir)